Add Security Considerations for SRT hash collision attack (#3005)
quicwg/base-drafts: draft-ietf-quic-transport.md

ID can be selected to route later packets to the same server.

## Hash Collision Attack on Stateless Reset Tokens

> As the stateless reset tokens are controlled by the peers, QUIC endpoints SHOULD

Reviewer: ```suggestion
Since stateless reset tokens are generated by peers, endpoints SHOULD
```

> be resistant to hash flooding DoS attacks when using a hash map for retaining

Reviewer: Is there an easy reference that can be cited for "hash flooding DoS attacks"?

> be resistant to hash flooding DoS attacks when using a hash map for retaining

Reviewer: ```suggestion
be resistant to hash flooding DoS attacks when using a hash map for storing and comparing
```

> those tokens. One way of achieving such property is to retain and compare the

Reviewer: ```suggestion
those tokens. One way of achieving this property is to store and compare
```

transformed values of the stateless tokens where the transformation is defined
as a pseudo-random permutation (e.g., block cipher) or a keyed hash (e.g., HMAC
{{?RFC2104}}) that is cryptographically secure, instead of using the raw token

> values as the hash keys.  This approach satisfies the secure comparison

Reviewer: Suggested rephrase: "... transformed values of the stateless tokens instead of the raw token values. A transformation such as a pseudo-random permutation (e.g. block cipher) or a cryptographically-secure keyed hash (e.g., HMAC {{?RFC2104}}) can be used to also ensure that comparing stateless tokens does not leak timing information, as specified in {{detect-stateless-reset}}."
